homeassistant/components/slide_local/config_flow.py

```diff
@@ -0,0 +1,129 @@
+"""Config flow for slide_local integration."""
+
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from goslideapi.goslideapi import (
+    AuthenticationFailed,
+    ClientConnectionError,
+    ClientTimeoutError,
+    DigestAuthCalcError,
+    GoSlideLocal as SlideLocalApi,
+)
+import voluptuous as vol
+
+from homeassistant.components.zeroconf import ZeroconfServiceInfo
+from homeassistant.config_entries import ConfigFlow, ConfigFlowResult
+from homeassistant.const import CONF_API_VERSION, CONF_HOST, CONF_MAC, CONF_PASSWORD
+from homeassistant.helpers.device_registry import format_mac
+from homeassistant.helpers.selector import (
+    BooleanSelector,
+    SelectOptionDict,
+    SelectSelector,
+    SelectSelectorConfig,
+    SelectSelectorMode,
+)
+
+from . import SlideConfigEntry
+from .const import CONF_INVERT_POSITION, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+BOOLEAN_SELECTOR = BooleanSelector()
+
+API_VERSION_SELECTOR = SelectSelector(
+    SelectSelectorConfig(
+        options=[
+            SelectOptionDict(value="1", label="API 1"),
+            SelectOptionDict(value="2", label="API 2"),
+        ],
+        mode=SelectSelectorMode.DROPDOWN,
+        translation_key=CONF_API_VERSION,
+    )
+)
```

Comment:
What's the difference? Can we auto detect this?

Reply:
There is a difference in the authentification. Not sure if other differences exist. The library needs the api version.

Based on https://www.reddit.com/r/slidecurtains/comments/1cwc5u2/which_firmware_version_does_your_slide_have/, API version should be 2, if zeroconf is enabled. Hence we could set it to 2 in this case. For the other cases I would leave it to the user, as according to @ualex73 (in doc to CC) one cannot determine the API version other than by trying out.

Reply:
What's bad with trying out? Assuming they are incompatible with each other I don't see a danger in just doing 2 attempts and if one work we store the right API version.

It would at least save the user from having to look up the version they have for that curtain

Reply:
I'm not sure what the response is, if the wrong API is used. So if we try API 2 and it does not work, it could be due to the API or anything else.

Reply:
You have a device and can try it out :)

Since its about auth, I expect (hope) that it just fails. If both API calls fail, we can just tell the user it does not work

Reply:
I only have API 2 devices, so I cannot test the failure of API 1. I guess the latter would be more interesting, as I would expect an Authentication error. Problem is, this could be due to wrong API or wrong password.
For API 2 devices, I think an Authentication error may indeed point to a wrong API (as basically no password should be required for API 2).
From an UX perspective I would prefer to document this (perhaps also with more extensive error messages) instead of (uncracefully) fail.

Reply:
Let's implement to our best knowledge and we can always see what happens with error messages

Reply:
Would you make the API an option then? Because I think somehow the user has to be able to change the version...

Reply:
I would say let's try to probe the connection. We should also be wary, what if someone updates their slide to V2? Is that even possible? Should these be runtime checks?

Reply:
Update is (or better was) possible over the cloud. Perhaps also directly, if you know how. Not sure how this will be in the future.
While I do like any testing/probing we can do, I think we also should give the user the possibility to configure.